data_safe_haven/config/config.py

```diff
@@ -260,6 +252,17 @@ def __init__(self, context: Context, **kwargs: dict[Any, Any]):
         tags = ConfigSectionTags(context)
         super().__init__(context=context, tags=tags, **kwargs)
 
+    @field_validator("sres")
+    @classmethod
+    def all_sre_indices_must_be_unique(
+        cls, v: dict[str, ConfigSectionSRE]
+    ) -> dict[str, ConfigSectionSRE]:
+        indices = [s.index for s in v.values()]
+        if len(indices) != len(set(indices)):
+            msg = "all SRE indices must be unique"
+            raise ValueError(msg)
+        return v
```

Comment:
Because this is quite similar to the database unique list validation, it might be possible to do something like create an Annotated Container Type `UniqueList[T]` and a validator for it.

Do you think that is possible? Let's open an issue if so and not block this.

Reply:
Sounds sensible, but I'm not sure enough of pydantic's inner workings to be sure.